src/getStepStartStates.ts

```diff
@@ -268,14 +271,50 @@ export function getResourceManagerSteps(): GetApiSteps {
   };
 }
 
-export default function getStepStartStates(
+export default async function getStepStartStates(
   executionContext: IntegrationExecutionContext<IntegrationConfig>,
-): StepStartStates {
+): Promise<StepStartStates> {
   const config = executionContext.instance.config || {};
-
-  const activeDirectory = { disabled: !config.ingestActiveDirectory };
-  const resourceManager = { disabled: !hasSubscriptionId(config) };
-  const managementGroups = { disabled: !config.configureSubscriptionInstances };
+  //Disable all the steps for subscriptions named 'Access to Azure Active Directory'
+  //These subscriptions do NOT host Azure AD. These are legacy subscriptions that can no longer be managed by customer portal. We decided not to execute any step, but just show an event.
+  //If any step executes we get: 'The current subscription type is not permitted to perform operations on any provider namespace. Please use a different subscription.'
+  //More info:
+  //https://www.jasonfritts.me/2020/04/07/what-is-the-access-to-azure-active-directory-subscription-for/#:%7E:text=The%20%E2%80%9CAccess%20to%20Azure%20Active%20Directory%E2%80%9D%20subscriptions%20are%20a%20legacy,portal.azure.com
+  const logger = executionContext.logger;
+  let disableAllSteps = false;
+  if (hasSubscriptionId(config)) {
+    const subscriptionClient = new J1SubscriptionClient(config, logger);
+    try {
+      const subscription = await subscriptionClient.getSubscription(
+        config.subscriptionId!,
+      );
+      if (
+        subscription &&
+        subscription.displayName == 'Access to Azure Active Directory'
+      ) {
+        disableAllSteps = true;
+        logger.publishInfoEvent({
+          name: IntegrationInfoEventName.Results,
+          description:
+            'The subscription is is not permitted to perform operations on any provider namespace. These are legacy subscriptions that can no longer be managed by customer portal. No steps will be executed.',
```

Comment:
I would suggest something like this:
```suggestion
          description:
            'This is an "Access to Azure Active Directory" subscription, which is a legacy subscription type that is no longer used. They were used prior to the current Azure Portal (https://portal.azure.com/). JupiterOne cannot ingest any resources from this subscription.',
```

Reply:
I'm also wondering if we should update our `configureSubscriptionInstances()` logic to stop ingesting these subscriptions to not even pull these in to begin with. 

```diff
function shouldSkipSubscription(subscription: Subscription) {
  return subscription.state !== 'Enabled' 
  || subscription.tags?.JupiterOne === 'SKIP'
+ || subscription.displayName === 'Access to Azure Active Directory'
}
```